File: C++/Programming Basics with C++/07. Exercise - Nested Conditional Statement/simple_task/FishingBoat.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
    int budget;
    string season;
    int numberFishermen;

    cin >> budget >> season >> numberFishermen;

    transform(season.begin(),
              season.end(),
              season.begin(),
              ::tolower);

    double rent = 0;

    if (season == "summer" || season == "autumn") {
        rent = 4200;
    } else if (season == "spring") {
        rent = 3000;
    } else if (season == "winter") {
        rent = 2600;
    }

    if (numberFishermen >= 1 && numberFishermen <= 6) {
        rent = rent - rent * 0.1;
    } else if (numberFishermen >= 7 && numberFishermen <= 11) {
        rent = rent - rent * 0.15;
    } else if (numberFishermen >= 12) {
        rent = rent - rent * 0.25;
    }
    if (season != "autumn" && numberFishermen % 2 == 0) {
        rent = rent - rent * 0.05;
    }

    double leftMoney = budget - rent;

    cout.setf(ios::fixed);
    cout.precision(2);

    if (leftMoney >= 0) {
        cout << "Yes! You have " << leftMoney << " leva left." << endl;
    } else if (leftMoney < 0) {

        cout << "Not enough money! You need " << abs(leftMoney) << " leva." << endl;
    }

    return 0;
}

File: C#/Programming Basics with C#/07. Nested Conditional Statements - Exercise/06. Fishing Boat/FishingBoat.cs
using System;

class FishingBoat
{
    static void Main()
    {
        int budget = int.Parse(Console.ReadLine());
        string season = Console.ReadLine().ToLower();
        int numberFishermen = int.Parse(Console.ReadLine());
        double rent = 0;

        if (season == "summer" || season == "autumn")
        {
            rent = 4200;
        }
        else if (season == "spring")
        {
            rent = 3000;
        }
        else if (season == "winter")
        {
            rent = 2600;
        }

        if (numberFishermen >= 1 && numberFishermen <= 6)
        {
            rent = rent - rent * 0.1;
        }
        else if (numberFishermen >= 7 && numberFishermen <= 11)
        {
            rent = rent - rent * 0.15;
        }
        else if (numberFishermen >= 12)
        {
            rent = rent - rent * 0.25;
        }
        if (season != "autumn" && numberFishermen % 2 == 0)
        {
            rent = rent - rent * 0.05;
        }

        double leftMoney = budget - rent;

        if (leftMoney >= 0)
        {
            Console.WriteLine($"Yes! You have {leftMoney:F2} leva left.");
        }
        else if (leftMoney < 0)
        {
            Console.WriteLine($"Not enough money! You need {Math.Abs(leftMoney):F2} leva.");
        }
    }
}